luaui/Widgets/cmd_blueprint.lua

```diff
@@ -240,6 +240,9 @@ for unitDefID, unitDef in pairs(UnitDefs) do
 	elseif unitDef.isBuilder and not unitDef.canMove and not unitDef.isFactory then
 		-- nanos
 		blueprintBuildableUnitDefs[unitDefID] = true
+	elseif unitDef.customParams.mine then
```

Comment:
I see other parts of the code also use `unitDef.customParams.detonaterange` or `unitDef.modCategories.mine`. Does it matter which is used?

Reply:
Which one of them is semantically "this unit def is a mine"?

Reply:
I could see `unitDef.customParams.mine` or `unitDef.modCategories.mine` could both fit that definition. I'm guessing you're implying `unitDef.modCategories.mine`?

Reply:
The `modCategories` one looks correct, though I can't find any documentation on this tag.

Reply:
`modCategories` are for targeting. Use a customparam, possibly making a new one if the existing one doesn't fit.

Reply:
I think that's correct, and that they are directly related to the "categories" field in unitDef files, based on these code sections:

* https://github.com/beyond-all-reason/spring/blob/0889e956f86847114f1944c931c519a2614b3872/rts/Lua/LuaConstGame.cpp#L230
* https://github.com/beyond-all-reason/spring/blob/0889e956f86847114f1944c931c519a2614b3872/rts/Sim/Units/UnitDef.cpp#L556

(which both use `CCategoryHandler`)

Reply:
> `modCategories` are for targeting. Use a customparam, possibly making a new one if the existing one doesn't fit.

Ah, thank you. Looks like there is a customparam, so I'll keep it using that.

Reply:
As a side note, why not just replace the whole block if `unitDef.isImmobile`?

Reply:
> As a side note, why not just replace the whole "building or nano or mine" block with `if unitDef.isImmobile`?

Mostly because I am not completely sure what `isImmobile` covers. I'll check out the list of what it applies to when I have a chance.

Reply:
ok, the units that have `isImmobile==true` and `isBuilding==false` are nanos, mines, base builders, Floating Plasma Deflectors, hats, and lootboxes.

Reply:
I'm going to resolve this conversation, as it sounds like the current solution is the best.